googleapiclient/http.py

```diff
@@ -1208,8 +1208,7 @@ def _serialize_request(self, request):
     if request.http is not None:
       credentials = _auth.get_credentials_from_http(request.http)
       if credentials is not None:
-        if _auth.has_access_token(credentials):
```

Comment:
damn, i should have tested my downstream usage of this fix before asking for #421. this `if` that @mgilson asked for in https://github.com/google/google-api-python-client/pull/376/commits/282fc889956e42de8ec3e26aecbe39509f027333#r109217447 was necessary to fix #350, at least in my case, and losing it here regresses and results in #350 happening for me again.

i've added it back in #434. apologies for not testing with this patch earlier!